runtime/onert/frontend/nnapi/wrapper/OperationFactory.cc

```diff
@@ -1415,6 +1419,8 @@ OperationFactory::OperationFactory()
 
     operation::ArgMax::Param param;
     param.axis = operands.at(OperandIndex{init_param.inputs[1]}).asScalar<std::int32_t>();
+    // NNAPI ARGMAX output type is always int32
+    param.output_type = DataType::INT32;
```

Comment:
It seems we should check this kind of specification handling for all other NNAPI operators as well. :)